platform/include/retry_utils.h

```diff
@@ -31,6 +31,141 @@
 /* Standard include. */
 #include <stdint.h>
 
+/**
+ * @page retryutils_page Retry Utilities
+ * @brief An abstraction of utilities for retrying with exponential back off and
+ * jitter.
+ *
+ * @section retryutils_overview Overview
+ * The retry utilities are a set of APIs that aid in retrying with exponential
+ * backoff and jitter. Exponential backoff with jitter is strongly recommended
+ * for retrying failed actions over the network with servers. Please see
+ * https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ for
+ * more information about the benefits with AWS.
+ *
+ * Exponential backoff with jitter is typically used when retrying a failed
+ * connection to the server. In an environment with poor connectivity, a client
+ * can get disconnected at any time. A backoff strategy helps the client to
+ * conserve battery by not repeatedly attempting reconnections when they are
+ * unlikely to succeed.
+ *
+ * Before retrying the failed communication to the server there is a quiet period.
+ * In this quiet period, the task that is retrying must sleep for some random
+ * amount of seconds between 0 and the lesser of a base value and a predefined
+ * maximum. The base is doubled with each retry attempt until the maximum is
+ * reached.<br>
+ *
+ * > sleep_seconds = random_between( 0, min( 2<sup>attempts_count</sup> * base_seconds, maximum_seconds ) )
```

Comment:
```suggestion
 * > sleep_seconds = random_between( 0, min( 2<sup>attempts_count</sup> * base_seconds, maximum_seconds ) )
 * >base_seconds = initial_seconds + random_between( 0, max_jitter ) 
```

Reply:
Sure but that is part of the code and not part of the simplified algorithm.